test/integration/sagemaker/test_spark.py

```diff
@@ -210,6 +210,66 @@ def test_sagemaker_pyspark_sse_s3(role, image_uri, sagemaker_session, region, sa
     assert len(output_contents) != 0
 
 
+def test_sagemaker_pyspark_sse_kms_s3(role, image_uri, sagemaker_session, region, sagemaker_client, account_id):
+    spark = PySparkProcessor(
+        base_job_name="sm-spark-py",
+        image_uri=image_uri,
+        role=role,
+        instance_count=2,
+        instance_type="ml.c5.xlarge",
+        max_runtime_in_seconds=1200,
+        sagemaker_session=sagemaker_session,
+    )
+
+    kms_key_id = None
+    kms_client = sagemaker_session.boto_session.client("kms", region_name=region)
+    for alias in kms_client.list_aliases()["Aliases"]:
+        if "s3" in alias["AliasName"]:
+            kms_key_id = alias["TargetKeyId"]
+
+    if not kms_key_id:
+        raise ValueError("AWS managed s3 kms key(alias: aws/s3) does not exist")
+
+    bucket = sagemaker_session.default_bucket()
+    timestamp = datetime.now().isoformat()
+    input_data_key = f"spark/input/sales/{timestamp}/data.jsonl"
+    input_data_uri = f"s3://{bucket}/{input_data_key}"
+    output_data_uri_prefix = f"spark/output/sales/{timestamp}"
+    output_data_uri = f"s3://{bucket}/{output_data_uri_prefix}"
+    s3_client = sagemaker_session.boto_session.client("s3", region_name=region)
+    with open("test/resources/data/files/data.jsonl") as data:
+        body = data.read()
+        s3_client.put_object(
+            Body=body, Bucket=bucket, Key=input_data_key, ServerSideEncryption="aws:kms", SSEKMSKeyId=kms_key_id
+        )
+
+    spark.run(
+        submit_app="test/resources/code/python/hello_py_spark/hello_py_spark_app.py",
+        submit_py_files=["test/resources/code/python/hello_py_spark/hello_py_spark_udfs.py"],
+        arguments=["--input", input_data_uri, "--output", output_data_uri],
+        configuration={
+            "Classification": "core-site",
+            "Properties": {
+                "fs.s3a.server-side-encryption-algorithm": "SSE-KMS",
+                "fs.s3a.server-side-encryption.key": f"arn:aws:kms:{region}:{account_id}:key/{kms_key_id}",
+            },
+        },
+    )
+    processing_job = spark.latest_job
+    waiter = sagemaker_client.get_waiter("processing_job_completed_or_stopped")
+    waiter.wait(
+        ProcessingJobName=processing_job.job_name,
+        # poll every 15 seconds. timeout after 15 minutes.
+        WaiterConfig={"Delay": 15, "MaxAttempts": 60},
+    )
+    output_contents = S3Downloader.list(output_data_uri, sagemaker_session=sagemaker_session)
+    assert len(output_contents) != 0
+    for s3_object in s3_client.list_objects(Bucket=bucket, Prefix=output_data_uri_prefix)["Contents"]:
+        object_metadata = s3_client.get_object(Bucket=bucket, Key=s3_object["Key"])
+        assert object_metadata["ServerSideEncryption"] == "aws:kms"
+        assert object_metadata["SSEKMSKeyId"] == f"arn:aws:kms:{region}:{account_id}:key/{kms_key_id}"
+
+
```

Comment:
It looks like you're doing two S3 lists: one when you do `output_contents = S3Downloader.list(...` and another when you do `s3_client.list_objects`.

If there aren't any objects in `s3_client.list_objects`, then the for loop won't run, so the test could silently and falsely succeed when it should fail.

Could you either:

1. Assert on the length of the `s3_client.list_objects` response, or
2. Use the `output_contents` you got from `S3Downloader.list`?